Afterpay countries

LeasoftKonstantinShapoval opened the pull request.

tiutiunnykleasoft committed:
Merge pull request #3 from emspay/development

Development

aleksandrkhruslovleasoft committed:
Merge pull request #5 from emspay/development

Development

shapovalkv committed:
Added hiding payment method tools for AfterPay with available countries in admin panel.

tiutiunnykleasoft commented on upload/admin/view/template/extension/payment/emspay.tpl:
```diff
@@ -281,6 +281,22 @@
+                            </span>
+                        </label>
+                        <div class="col-sm-10">
+                            <input type="text" name="ems_country_access" id="input-ems-country-access"
```
Что такое ems_country_access, почему нигде не обьявляешь.
На странице чекаута из-за этого не корректно выводится 
![изображение](https://user-images.githubusercontent.com/60009163/86147758-737b2380-bb02-11ea-8ed4-ec490786d652.png)

LeasoftKonstantinShapoval replied:
Пофиксил.

tiutiunnykleasoft commented on upload/admin/controller/extension/payment/emspay_ideal.php:
```diff
@@ -129,6 +131,7 @@ protected function getTemplateData()
             'entry_sort_order' => $this->language->get('entry_sort_order'),
             'entry_status' => $this->language->get('entry_status'),
             'entry_ems_total' => $this->language->get('entry_ems_total'),
+            'entry_country_access' => $this->language->get('entry_country_access'),
```
В файлах перевода нету таких полей, добавь переводы

LeasoftKonstantinShapoval replied:
Добавил локализацию.

tiutiunnykleasoft commented on upload/admin/controller/extension/payment/emspay_ideal.php:
```diff
@@ -117,6 +118,7 @@ protected function getTemplateData()
             'info_help_klarna_test_api_key' => $this->language->get('info_help_klarna_test_api_key'),
             'info_help_afterpay_ip_filter' => $this->language->get('info_help_afterpay_ip_filter'),
             'info_help_afterpay_test_api_key' => $this->language->get('info_help_afterpay_test_api_key'),
+            'info_help_country_access' => $this->language->get('info_help_country_access'),
```
Тут тоже нужно добавить в файлы переводов это поле

LeasoftKonstantinShapoval replied:
Добавил локализацию.

tiutiunnykleasoft commented on upload/catalog/model/extension/payment/emspay_afterpay.php:
```diff
@@ -25,15 +24,11 @@ public function getMethod($address, $total)
- 
-        if (!in_array(strtoupper(($this->session->data['payment_address']['iso_code_2'])), static::$apAllowedLocales)) 
-        {
-            return [
```
В этом Return возврашалось сообщение о том что AfterPay не поддерживается, почему решил его убрать?

LeasoftKonstantinShapoval replied:
Изходя из логики. Нет необходитьмости выводить данный текст.

tiutiunnykleasoft commented on upload/catalog/model/extension/payment/emspay_afterpay.php:
```diff
@@ -25,15 +24,11 @@ public function getMethod($address, $total)
-        {
-            return [
-                'code' => 'emspay_afterpay',
-                'title' => $this->language->get('text_title').' ('.$this->language->get('info_afterpay_not_allowed').')',
```
Почему не убрал с переводов это поле : `info_afterpay_not_allowed` если удалил его вывод?

LeasoftKonstantinShapoval replied:
Удалил.

tiutiunnykleasoft commented on upload/catalog/model/extension/payment/emspay_afterpay.php:
```diff
@@ -25,15 +24,11 @@ public function getMethod($address, $total)
-                'sort_order' => $this->config->get('emspay_afterpay_sort_order')
-            ];
+
+        if (!EmsHelper::CountryValidator($this->config->get('emspay_afterpay_country_access'),
```
No new line transafered is needed here

LeasoftKonstantinShapoval committed:
Updated AfterPay country available.
Added localization for AfterPay.
Updated version for 1.5.1 and add README.md and CHANGELOG.md description.

tiutiunnykleasoft commented on upload/system/library/emshelper.php:
```diff
@@ -657,6 +657,22 @@ public static function ipIsEnabled($ipList)
+     */
+    public static function CountryValidator($countryList, $billingAddress)
+    {
+        if ($countryList !== '' ) {
```
use `strlen()` or `empty() `function instead of `!== `

LeasoftKonstantinShapoval replied:
Done.

tiutiunnykleasoft commented on upload/system/library/emshelper.php:
```diff
@@ -657,6 +657,22 @@ public static function ipIsEnabled($ipList)
+     * @param $countryList
+     * @return bool
+     */
+    public static function CountryValidator($countryList, $billingAddress)
```
spell function name from the first small letter: like country Validator

LeasoftKonstantinShapoval replied:
Done

tiutiunnykleasoft commented on upload/catalog/model/extension/payment/emspay_afterpay.php:
```diff
@@ -25,17 +23,13 @@ public function getMethod($address, $total)
+
+        if (!EmsHelper::CountryValidator($this->config->get('emspay_afterpay_country_access'),
+            $this->session->data['payment_address']['iso_code_2'])) {
+
```
remove empty string there

LeasoftKonstantinShapoval replied:
Done

tiutiunnykleasoft commented on upload/admin/language/de-de/extension/payment/emspay_afterpay.php:
```diff
@@ -0,0 +1,58 @@
+<?php
```
Why this file is marked as 'new'? 
They must be 'changed' not the 'new'

tiutiunnykleasoft commented on upload/admin/language/fr-fr/extension/payment/emspay_afterpay.php:
```diff
@@ -0,0 +1,58 @@
+<?php
```
Why this file is marked as 'new'? 
They must be 'changed' not the 'new'

tiutiunnykleasoft commented on upload/admin/language/nl-nl/extension/payment/emspay_afterpay.php:
```diff
@@ -0,0 +1,58 @@
+<?php
```
Why this file is marked as 'new'? 
They must be 'changed' not the 'new'

tiutiunnykleasoft reviewed (commented):
All comments fixed correctly, added some spelling moment which doesn't affecting logic of hiding payment method.

tiutiunnykleasoft commented on upload/admin/view/template/extension/payment/emspay.tpl:
```diff
@@ -281,6 +281,22 @@
                         </div>
                     </div>
 
+                    <?php if ($emspay_module == 'emspay_afterpay'): ?>
```
This file already has `if block` for adding a custom testing data for Afterpay payment method, use that block, don't create a new block like this.

LeasoftKonstantinShapoval replied:
Removed.

LeasoftKonstantinShapoval committed:
Refactored code:
- Added logic for emty "Countries available for AfterPay" field.
- Transferred AfterPay countries availability field in emspay.tpl
- Removed all empty string.
- Changed localization for AftePay countries availability.

LeasoftKonstantinShapoval committed:
Refactored return in countryValidator

LeasoftKonstantinShapoval committed:
Delete needless return
